kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb

```diff
@@ -131,24 +142,43 @@ def get_file_name(coordinates)
           end
         end
 
-        def get_query_params(coordinates)
+        def get_query_params(coordinates, what_parameters = nil)
           artifact = parse_coordinates(coordinates)
-          @version = artifact[:version]
+          @version = artifact[:version].to_s.upcase
+
           query = {:g => artifact[:group_id], :a => artifact[:artifact_id], :e => artifact[:extension], :v => version, :r => configuration[:repository]}
           query.merge!({:c => artifact[:classifier]}) unless artifact[:classifier].nil?
-          query
+
+          params = what_parameters.nil? ? query : Hash.new
+          what_parameters.each {|key| params[key] = query[key] } unless what_parameters.nil?
+
+          params.map{|key,value| "#{key}=#{value}"}.join('&')
```

Comment:
Might not be strictly needed, but it's good practice to URI-encode the parameters: `"#{CGI.escape key}=#{CGI.escape value}"`.